### PR TITLE
feat: use wagmi address resolution hooks/utils

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR uses the [wagmi](https://wagmi.sh/) library for ENS reverse/address resolution.

- Uses [`useEnsName`](https://wagmi.sh/docs/hooks/useEnsName) for reverse resolution in components
- Uses [`useEnsAddress`](https://wagmi.sh/docs/hooks/useEnsAddress) for resolution in components
- Removes our in-house logic in favor of `fetchEnsName` for reverse resolution in pure functions (from `@wagmi/core`, not documented as hooks are the recommended use of wagmi)
- Removes our in-house logic in favor of `fetchEnsAddress` for resolution in pure functions (from `@wagmi/core`, not documented as hooks are the recommended use of wagmi)

This also refactors / cleanup / shuffles things a bit to accomodate for the use of these hooks.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

closes #

## Risk

Possible regressions across User Menu, fiat ramps, and receive modal, test accordingly.

## Testing

- UserMenu should show no regressions for ENS, addresses, and labels
- Transaction history should show no regressions for ENS and addresses
- Fiat ramps should show no regressions for ENS and addresses
- Receive modal should show no regressions for ENS and addresses

### Engineering

- Refer to top-level testing notes

### Operations

- Given this touches the domain of addresses, send and receives should be tested.

### CODE REVIEW

File: src/components/Modals/FiatRamps/views/Manager.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const ManagerRouter: React.FC<ManagerRouterProps> = ({ fiatRampProvider }) => {
   const [bchAddress, setBchAddress] = useState<string>('')
   const [dogeAddress, setDogeAddress] = useState<string>('')
   const [ltcAddress, setLtcAddress] = useState<string>('')
-  const [ethAddress, setEthAddress] = useState<string>('')
+  const [ethAddress, setEthAddress] = useState<string | undefined>()
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/views/Manager.tsx:76]: Allows the query to be skipped

https://wagmi.sh/docs/hooks/useEnsName
> If address is not defined, hook will not run

File: src/components/Modals/Receive/ReceiveInfo.tsx [hunk 2]
```diff
@@ -59,6 +59,13 @@ export const ReceiveInfo = ({ asset, accountId }: ReceivePropsType) => {
   const accountNumber = useAppSelector(state => selectAccountNumberByAccountId(state, filter))
   const { utxoParams, accountType } = accountIdToUtxoParams(accountId, accountNumber ?? 0)
 
+  const { data, isLoading, isSuccess } = useEnsName({
```
Comment (gomesalexandre) [inline src/components/Modals/Receive/ReceiveInfo.tsx:62]: Rename

File: src/context/WagmiProvider/WagmiProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { getConfig } from 'config'
+import last from 'lodash/last'
+import type { Chain } from 'wagmi'
+import { chain, configureChains, createClient, WagmiConfig } from 'wagmi'
+import { infuraProvider } from 'wagmi/providers/infura'
+import { jsonRpcProvider } from 'wagmi/providers/jsonRpc'
+
+const avalancheChain: Chain = {
```
Comment (gomesalexandre) [inline src/context/WagmiProvider/WagmiProvider.tsx:8]: Find a better place and/or make programmatic? See

https://github.com/shapeshift/web/pull/2348/files#diff-c2977d04ff57b046d9667291f82e582192401a3d048c937cc77bee3bc1128440R79-R88

File: src/components/Modals/Receive/ReceiveInfo.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/Modals/Receive/ReceiveInfo.tsx]: d4971d603b5b3c6f6e33e0ed046f215b834d04c3

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2701** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2701" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2701?utm_source=stack-comment).

File: src/context/WagmiProvider/WagmiProvider.tsx (hunk 3 above)
Comment (gomesalexandre) [inline src/context/WagmiProvider/WagmiProvider.tsx:8]: Note: [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) `AddEthereumChainParameter`-ish compliant, with a few discrepancies

Comment (gomesalexandre) [reply src/context/WagmiProvider/WagmiProvider.tsx:8]: Probably the best to leave it here as a const for now, programmatic params getting would be possible but require more work that would better live in its own PR

**Review by 0xApotheosis — APPROVED** [review]
Can't fully test due to Infura `429`ing, but the code looks sane and it appears to be making the ENS requests as expected when running locally.

Approving with the hope that this significantly reduces Infura requests.

**Pull request merged**